src/main/java/com/google/api/codegen/discogapic/transformer/DiscoGapicParser.java

```diff
@@ -142,20 +128,34 @@ public static Name methodAsName(Method method) {
 
   /**
    * Return the name of the fully qualified resource from a given canonicalized path. Use {@link
-   * #getCanonicalPath(Method)}} for canonicalization of the parameter.
+   * #getCanonicalPath(String)}} for canonicalization of the parameter. This method includes all
+   * wildcards from the input path in the resulting Name, and it includes literal segments from the
+   * input path
    */
   public static Name getQualifiedResourceIdentifier(String canonicalPath) {
     String[] pieces = canonicalPath.split(PATH_DELIMITER);
 
     Name name = null;
-    for (String piece : pieces) {
-      if (!piece.contains("{")) {
-        if (name == null) {
-          name = Name.from(Inflector.singularize(piece));
-        } else {
-          name = name.join(stringToName(Inflector.singularize(piece)));
+    for (int i = 0; i < pieces.length; i++) {
+      if (i < pieces.length - 1 && pieces[i + 1].contains("{")) {
+        // Current index at a literal segment directly preceding a wildcard segment.
+        // Skip literal segment iff it is the same as or the pluralized version of the wildcard string.
```

Comment:
I may be reading this wrong, but if we skip the segments that precede `{bracketed}` segments, isn't that the same as singularizing the preceding segment then skipping the bracketed segment, which is what the original code is doing? The only difference is when the preceding segment is not a pluralized version of the bracketed segment, when we include both. Is that what we want?

If that is what we want, then let's always include the unbracketed segment, and skip the bracketed one when it matches the previous. This lets us avoid indexing and lookahead. We could even just always check for duplication, so something like:
```
Name name = Name.from();
String previous = null;
for (String piece : pieces) {
  String next = Inflector.singularize(piece);
  if (!next.equals(previous)) {
    // Only append to the name if this segment is not identical to the previous segment
    name = name.join(stringToName(next));
    previous = next;
  }
}
```
Does that work?

Reply:
Yes, thanks so much!